### ISSUE TITLE
Review of existing solutions (platforms and individual components to bolt together)

### ISSUE THREAD

**Issue by yarikoptic** [con/ference#1]
the idea came up while searching for a name for the OHBM virtual posters setup (now called GallOP): https://github.com/datalad-datasets/ohbm2020-posters/issues/135#issuecomment-652486445, and then briefly discussed in the OHBM OSR crowdcast on establishing an **open source conferencing platform** which would fit wide range of needs.  Here ATM I would like not really to carry the discussion of what we should aim for but research components and already existing platforms which could provide a backbone for the con/ference solution which could be used for various online and/or hybrid conferences.

### Integration platforms:

- [GallOP](https://github.com/datalad-datasets/ohbm2020-posters) - visit https://datalad-datasets.github.io/ohbm2020-posters/ .  Minimalistic quick&dirty solution with jitsi + etherpad rooms per each poster
- [DebConf (Debian Conference) 19 platform](https://salsa.debian.org/debconf-team/public/websites/dc19) - visit https://debconf19.debconf.org . DebConf AFAIK has been a pioneer in open "hybrid" conferencing: online casting + IRC chats since forever.  Thanks to the DebConf video team videos has been provided for a good number of years from https://meetings-archive.debian.net/pub/debian-meetings/ (git-annex of it is hiding somewhere)
- [BrainWeb](https://brain-web.github.io/) - ongoing effort by @r03ert0 and @katjaq to establish "a permanent virtual space for online collaborations". 
- [MiniConf](https://github.com/Mini-Conf/Mini-Conf) (thanks @emdupre) - "manages the papers, schedules, and speakers for an academic conference run virtually. It can be easily integrated with interactive tools such as video, chat, and QA". ([I chimed in](https://github.com/Mini-Conf/Mini-Conf/issues/78))
- ... ? ...

### Individual tech

- [Jitsi](https://jitsi.org/) became a savior of OHBM 2020 posters sessions.  We used central free https://meet.jit.si/ deployment which is good for small (<20) groups. For large groups or casting jitsi provides various additional components: it cast cast to Youtube, or https://jitsi.org/jitsi-videobridge/ could be deployed for server-side mixing (like zoom);  paid support could be sought after in the jitsi community https://community.jitsi.org/c/users/paid-work
- [Calla (formerly lozya)](https://github.com/capnmidnight/Calla) -- gather.town like environment, 1 man fun project, which I discovered before I knew about gather.town.  It is much leaner overlay on top of jitsi than gather.town. That might have its pros (more convenient larger video screens, easier to integrate with other tech, etc).
- [Mozilla hubs](https://hubs.mozilla.com/#/) -- an alternative for socializing environment (thanks @katjaq and @satra IIRC for referencing).  Yet to investigate in detail.
- ... ? ...

### External (closed source) services we liked

- [crowdcast](https://www.crowdcast.io/) for casting talks/discussions.  See pros/cons (#2), open alternatives: #3
- [gather.town](https://gather.town/) for socials. See Calla above as for an open (less polished yet) alternative
- [zoom](https://zoom.com) provided superior audio quality needed for disco events, and also better than default meet.jit.si for video conferences with > 10 participants (could be mitigated, issue TODO)

### External services worth delegating to

Instead of deploying and maintain service servers and CDN (would cost money anyways), some aspects probably better to delegate to external services to pay for the duration of the conference:

- **Video Casting**
  - https://vimeo.com/ - Analysis TODO in particular to the parallel rooms etc
  - ~~https://www.youtube.com/~~ since in case of any hint of copyright infringement the whole channel could be blocked jeopardizing the entire meeting. Also probably would be proliferated with Ads.  Could be used though as a supplementary sink for the casts and archival.
  - See above about DebConf video setup

Also attn to @vsoch @nicholst @effigies @rmarkello @adswa @bpoldrack @emdupre @soichih ;)

**Comment by emdupre** [comment]
For existing integration platforms, I'd also point to Mini-Conf ! https://github.com/Mini-Conf/Mini-Conf

**Comment by yarikoptic** [comment]
Wow -- a great new (was developed for  ICLR 2020 with 6000 participants!) find - added to the list above!  If I only knew two weeks back!!!

**Comment by yarikoptic** [comment]
some references to investigate from debconf ppl, but some aren't open (e.g. Easychair)

```
13:02   gwolf: In 2014 we switched to... a conference management system used (then, at least) by Ubuntu. I don't recall the name
13:02   gwolf: In more academic settings, I have used Open Conference Systems (OCS), by the Public Knowledge Project (PKP). I cannot tell you I like it, but it's very widely used
13:03   gwolf: Also for academic settings, I have seen many organizers prefering Easychair (although I don't know if it's available to be self-hosted or it's just a non-free SaaS)
13:03   terceiro: before wafer debconf used one called summit, wasn't it?
13:04   terceiro: it's even still online: https://summit.debconf.org/debconf15/ https://summit.debconf.org/debconf14/
13:06   terceiro: yoh: also https://github.com/topics/conference-management
13:07   terceiro: wafer is probably not even in that list
```

**Comment by soichih** [comment]
This is a paid service, but I've attended a few conferences that use https://attendify.com/ I thought the experience was pretty good. I think OHBM could let companies like this to take care of the back-bone of the conference experience (registration, communication, etc) as long as we can plugin various other services/*apps* that are nice to have.. and fun to work with?

**Comment by katjaq** [comment]
Hello! thanks for setting this up here! :) 🌟 
I think [gather town](https://gather.town) should definitely be added to the list above – just like jitsi this was our socialising hero platform! ❤️ 😍 
I love gather town! Maybe we could also add [Mozilla Hubs](https://hubs.mozilla.com/#/) to the list above :) Just to keep it in mind as another cute socialising platform option, based on the same principle of audio radius for interaction, and while gather town is 2D (lovely!), Mozilla Hubs is made for VR :D and is a cute low-poly environment to walk around an avatar in space (super sweet even without VR glasses ;) (there are videos on Youtube in case you want to quickly get a vis impression :)
I also like very much @soichih 's jitsi-etherpad integration ✨ So we could also add [etherpad](https://etherpad.org) to the list above if you like.

Looking forward to making this happen with all of you 😍 🚀

**Comment by yarikoptic** [comment]
Thank you @katjaq !

> I think [gather town](https://gather.town) should definitely be added to the list above

Unfortunately I can't, even though I loved it and appreciate all the work gather.town team has put in to make OHBM social gatherings a success. I have now put in bold the requirement for the solution to be open source, and that is why I have added Calla originally as the alternative.  Mozilla Hubs and Etherpad - will add for sure, thanks, forgot to do so in the "rush" ;)

**Comment by effigies** [comment]
Would it be worth having a feature list of closed-source tools we've used that we liked? For example, while I don't think crowdcast was perfect, it was an improvement over the conference site and it's not clear we can achieve those improvements with a listed FOSS component.

**Comment by yarikoptic** [comment]
Ok, I will add "external (closed source) services" section with gather town and crowdcast. I still think though that pretty much for any of them we could find/establish an open source alternative so if contributed to, it could provide needed functionality. I would also prefer to have open source solution and just pay for maintenance, infrastructure, support.

**Comment by effigies** [comment]
Agreed. This is just to help focus our searches with explicit criteria.

**Comment by effigies** [comment]
Agreed. This is just to help focus our searches with explicit criteria.

**Comment by yarikoptic** [comment]
Started https://github.com/con/ference/issues/2 from a participant perspective. There could be pros I don't know about from crowdcast sessions organizer perspective. Please contribute there

**Comment by robertoostenveld** [comment]
Colleagues of mine working in developmental neuroscience are currently (i.e. this week) attending the yearly ICIS conference, which is now virtual. See https://infantstudies.org/congress-2020/. It makes use of this service https://event.vconferenceonline.com and somehow is related to this https://podiumconferences.com (the latter might be the bureau organizing it). I have been able to look along (to some limited extent). 

There is a single website (works best in chrome or edge, not firefox), with an agenda for different timezones (although I believe that all live discussions are only happening once), there is a virtual meeting hall (with pre-recorded video), there is a virtual poster- and exhibition hall (also including commercial exhibitors, some of whom have a zoom link there, some have a text chat only), there are poster sessions (where each presenter organizes his/her own section, some have a zoom link, some have a pre-recorded video run-through of the poster, some only text chat) and rooms for the symposia (with a pre-recorded welcome from the session chair, followed by pre-recorded lecture videos, followed by a live Q&A). 

A nice feature is that in the poster sessions (and I presume also in the symposia) you can see who is logged in and attending a specific session/poster, i.e. there is a list with names, positions, affiliations, and contact details. 

A problematic feature is that it did not work with all browsers, and not everyone could get the video lectures to work (probably global bandwidth issues).  Personally I did not like that the chat functionality was scattered around, i.e. each page (which represents a persistent section of the program) seemed to have its own chat.

**Comment by TomEmotion** [comment]
How do commercial offerings fit in here, even if just as a demonstration of what might be possible to develop in open source? Here is Australia, people have told me about iSee, https://www.iseevc.com.au/, which appears to offer 3D functionality with video and proximity audio, as well as a bunch of other features.

**Comment by soichih** [comment]
I think commercial offerings are not the solution that we might ultimate adopt, but I think we can learn from, and used as inspiration. So we should at least bring them up and discuss pros/cons? I recently attended PEARC20 and they used brella.io. It was a simple yet quite effective with keeping up with all the events and had good integration with Jitsi (you can request / schedule impromptu meetings)

brella.io also inspired me to frame "con/ference" solution as a "calendar" or event management problem. We need to know what is(will be) happening, when, and how to connect to it. We need to know which event people are currently attending (and will be attending), and anyone should be able to propose a meeting (either group or private), find available time for all attendees. I am thinking we can develop a similar UI used for video/audio editing where "tracks" representing virtual "meeting place" and X-axis represents time.

Also, this is just my personal opinion, but trying to create a platform that tries to simulate a physical conference space via 3D or 2D type virtual environment is not the most optimal way to do this - as virtual/remote conference is fundamentally different and governed by very different constraints from a real conference. For one thing, people will be attending from many different time zones, so it is probably more important to provide a good way to manage "shared time" rather than "shared space". Also, in a virtual conference, we can take advantage of many different technologies (web browser, offline video, online VC, database, etc..) I understand the urge to make it look and feel more like a physical conference space, but I think doing so would ultimately limit the capability of con/ference.

**Comment by yarikoptic** [comment]
> trying to create a platform that tries to simulate a physical conference space via 3D or 2D type virtual environment is not the most optimal way to do this

+1 on that! e.g. I briefly tried out mozilla hub 3d environments - cute but it is would just making entire experience really inefficient.  For social interactions I really liked the idea of gather.town or calla more since it actually is not imposing all the difficulties with navigation as much and actually makes it "lazy fun". But I do not even see much benefit from any "physical space" pretense/navigation in the context of talks/posters/etc.
  
With that in mind -- and what we had done for GallOP -- having an ability to overview "where the people are ATM" is IMHO a great feature to have... Unlike in real life it could also add flavor to it similarly to what neuromatch did -- show where "people of similar interests etc" are or even similar to `gather.town` to show where any particular attendee is (although not 100% certain about that - we would like to encourage stalking etc, but also could have a blacklist or smth ;))

**Comment by vsoch** [comment]
I think we need to take caution with these commercial environments. I'm probably just hugely sensitive about sharing personal information, I attended PEARC and also briefly jumped into the brain web / jitsu / gather town and found it much more fun, and actually akin to how you run around a conference and find people. Brella is like watching a webinar, and then at most having chat windows (so it's more akin to Zoom, but harder to figure out where things are and you can't present your screen easily for say, a poster presentation). More importantly, it's a third party service that asks for fairly significant personal information. If you read the privacy notice they will freely share/transfer data with 3rd parties:

>  We may transfer your personal data to third parties, as it is a normal course of doing business in a digitalized world. When personal data is transferred to third parties, we ensure that we conclude adequate personal data processing agreements and safeguards in relation to the data transfers.

I'm not sure about others, but I don't like the idea of forcing participants to agree to such terms, or too bad, you can't be a part of the event. It's worth exploring more strongly avenues that don't require this sharing of personal data, or even other third party services that allow to join without becoming a customer. We should be empowered to be in control of our personal information, and not have to make a choice to share or not be able to participate at all.

**Comment by TomEmotion** [comment]
Wow - that's a really vague privacy notice! Nowhere near acceptable, as far as I'm concerned. I agree that data privacy is incredibly important.

On the issue of simulated 2d/3d environments and whether they have a role to play in online conferences, I think it's very context, and user dependent. Given the variety of different presentation formats possible, and the variety of communication objectives and personal styles and preferences, it would be nice to offer multiple options to conference attendees.

It is also worth considering that for many people, sitting staring at a computer screen for a prolonged period of time is exhausting. It needs to be broken up, not only with breaks, but also with the type of interactive environment.

Personally, I am a very visuospatially organised person. The graphical desktop was a great metaphor for people like me because it is based on the same spatial configuration as a desktop. Likewise, something that allows me to orient to different parts of a conference spatially is enormously helpful, non-fatiguing and comfortable. Others will be different - some will prefer a timeline, others a calendar style system, and others an interface more like a mobile device. The point is, for a conference with a variety of attendees, there needs to be consideration for these aspects, otherwise people will be excluded.

I know that's asking for a lot, and I don't have the answers. But I hope it's a useful perspective.

**Comment by r03ert0** [comment]
I was also thinking about the desktop metaphor. There used to be quite some discussion about it at the time mac/windows wars, and I imagine also some ergonomy research? Probably mmorpgs are a successful example of interface where people communicate and work together in a virtual space? At the moment, for our community, it seems to me the main function of 2D gather.town has been to be funny, so that you prefer to go there and see people instead of doing something more productive elsewhere.

**Comment by patcon** [comment]
Hi all! Came across your list and noticed that I've been using one that hasn't been mentioned yet: https://www.highfidelity.com

I've used it for (or helped friends use it for):
- multi-lingual birthday parties
- civic hacknights (events with breakouts)
- general socializing
- giving presentations
- combined meeting/coworks

Here are my notes, in case it's helpful! Thanks for sharing your own internal conversations (I'll credit anything I use in an upcoming [tech worker co-op show & tell talk](https://agaric.coop/show-and-tell) I'm preparing for :) 
https://hackmd.io/@patcon/SybjaiSbD

**Comment by patcon** [comment]
Oh hey, and another write-up about organizing the alife conference which I thought might be helpful (sorry, if spammy!)
https://medium.com/@juniper.lovato/a-how-to-reflections-on-planning-virtual-science-conferences-eeb754ed404b

**Comment by katjaq** [comment]
Thank you @patcon 🎉  Don't worry, this is what the list is for :)  Thank you very much for sharing your experience and notes, and adding tools we hadn't come across 🚀 . Good luck/enjoy your talk. In case it will be recorded, feel invited to drop us a link here :)

**Comment by patcon** [comment]
It's an open call this Wed at 11am ET / 15:00 UTC, if anyone wishes to join! Should be recorded too:
https://lists.mayfirst.org/pipermail/showandtell/2020-August/000156.html

**Comment by yarikoptic** [comment]
just a note: recording for that showandtell is [available](https://na66.distancelearning.cloud/playback/presentation/2.0/playback.html?meetingId=9db03309b42d637e676c1417bdc165b466df2da8-1599057250718) (yet to watch myself)

**Comment by helenvarley** [comment]
https://bigbluebutton.org/

**Comment by patcon** [comment]
https://www.meet.coop/about/ hosts big blue button, and runs as a democratic #platformcoop, owned partially by users and by workers. Mentioning in case that sort of conference platform aligns with any strongly held values of your group :)

Disclaimer: I work at a web shop that's run as a non-profit worker co-operative, and we are founding members of Meet.coop

**Comment by brylie** [comment]
[WorkAdventure](https://workadventu.re/) is an open-source virtual venue/office space. They also offer free hosting for open-source projects.

**Comment by Ljon4ik4** [comment]
[Comingle](https://github.com/edemaine/comingle) is a meeting platform seemingly multiple sessions/ threads.
Its video is based on jitsi, there is also a shared whiteboard option (cocreate) which is quite nice.

**Comment by nicopace** [comment]
> [WorkAdventure](https://workadventu.re/) is an open-source virtual venue/office space. They also offer free hosting for open-source projects.

WorkAdventure is not open source :(

You can find information about the license here: https://workadventu.re/faq

It says:
> The self-hosted version of WorkAdventure is distributed under the AGPL v3 license modified by the Commons Clause.

That means that it is not Open source, as they state it in that same page.

**Comment by vsoch** [comment]
AGPL just means if you make changes you have to also release them under the same license. And the commons clause they add just means you can’t sell it. For the purposes of the group here it may not be OSI compliant but I would still consider it useful for the group here.

**Comment by nicopace** [comment]
> AGPL just means if you make changes you have to also release them under the same license. And the commons clause they add just means you can’t sell it. For the purposes of the group here it may not be OSI compliant but I would still consider it useful for the group here.

Not being OSI complaint means that you can't combine it with other code that is AGPL, because of the Commons clause :(
Sure, it can be considered... I just wanted to point out that it might not be useful if you consider integrating it into other tools, or contributing to it by importing another library.

**Comment by vsoch** [comment]
Yeah that’s a good point, most tech companies etc., do not like that license.

**Comment by nicopace** [comment]
The hard part is the commons clause, AGPL is fine as long as it is just that... but the commons clause makes in impossible to combine.
